arangod/IResearch/IResearchLink.cpp

```diff
@@ -1401,7 +1418,7 @@ arangodb::Result IResearchLink::insert( // insert document
   if (!trx.state()) {
     return arangodb::Result(
       TRI_ERROR_BAD_PARAMETER,
```

Comment:
I think this is not a `TRI_ERROR_BAD_PARAMETER` (as this is normally used for bad parameters submitted by end users) but this justifies a `TRI_ERROR_INTERNAL` (an error that is mostly likely due to a programming error that the end user can do nothing about).